x-pack/plugins/uptime/server/lib/saved_objects/saved_objects.ts

```diff
@@ -9,33 +9,43 @@ import {
   SavedObjectsErrorHelpers,
   SavedObjectsServiceSetup,
 } from '../../../../../../src/core/server';
+import { EncryptedSavedObjectsPluginSetup } from '../../../../encrypted_saved_objects/server';
+
 import { DYNAMIC_SETTINGS_DEFAULTS } from '../../../common/constants';
 import { DynamicSettings } from '../../../common/runtime_types';
 import { UMSavedObjectsQueryFn } from '../adapters';
 import { UptimeConfig } from '../../config';
 import { settingsObjectId, umDynamicSettings } from './uptime_settings';
 import { syntheticsMonitor } from './synthetics_monitor';
-
-export interface UMSavedObjectsAdapter {
-  config: UptimeConfig;
-  getUptimeDynamicSettings: UMSavedObjectsQueryFn<DynamicSettings>;
-  setUptimeDynamicSettings: UMSavedObjectsQueryFn<void, DynamicSettings>;
-}
+import { syntheticsServiceApiKey } from './service_api_key';
 
 export const registerUptimeSavedObjects = (
   savedObjectsService: SavedObjectsServiceSetup,
+  encryptedSavedObjects: EncryptedSavedObjectsPluginSetup,
   config: UptimeConfig
 ) => {
   savedObjectsService.registerType(umDynamicSettings);
 
   if (config?.unsafe.service.enabled) {
     savedObjectsService.registerType(syntheticsMonitor);
+    savedObjectsService.registerType(syntheticsServiceApiKey);
+
+    encryptedSavedObjects.registerType({
+      type: syntheticsServiceApiKey.name,
+      attributesToEncrypt: new Set(['apiKey']),
```

Comment:
As I mentioned above, when I was testing this the saved object service also returned to me an `encoded` key, which was the pre-encoded `id:apiKey` combination. If that's the case, we should also encrypted that key, or only save the `encoded` key on the saved object and encrypted that.